### ISSUE TITLE
Unable to load saved model

### ISSUE THREAD

**Issue by prashantmaheshwari94** [tensorflow/tensorflow#12150]
I am facing similar issue but for GRU. I am using tensorflow  1.1.0 and I tried dumping the model in different ways: 
a)          saver = tf.train.Saver(tf.global_variables())
        model_exporter = exporter.Exporter(saver)

        # Restore variables from training checkpoint
        # TODO: This restores the most recent checkpoint, but if we use validation to counterract
        #       over-fitting, we may want to restore an earlier checkpoint.
        checkpoint = tf.train.get_checkpoint_state(FLAGS.checkpoint_dir)
        checkpoint_path = checkpoint.model_checkpoint_path
        saver.restore(session, checkpoint_path)
        log_info('Restored checkpoint at training epoch %d' % (int(checkpoint_path.split('-')[-1]) + 1))

        # Initialise the model exporter and export the model
        model_exporter.init(session.graph.as_graph_def(),
                            named_graph_signatures = {
                                'inputs': exporter.generic_signature(
                                    { 'input': input_tensor,
                                      'input_lengths': seq_length}),
                                'outputs': exporter.generic_signature(
                                    { 'outputs': decoded})})
        if FLAGS.remove_export:
            actual_export_dir = os.path.join(FLAGS.export_dir, '%08d' % FLAGS.export_version)
            if os.path.isdir(actual_export_dir):
                log_info('Removing old export')
                shutil.rmtree(actual_FLAGS.export_dir)
        try:
            # Export serving model
            model_exporter.export(FLAGS.export_dir, tf.constant(FLAGS.export_version), session)

            # Export graph
            input_graph_name = 'input_graph.pb'
            tf.train.write_graph(session.graph, FLAGS.export_dir, input_graph_name, as_text=False)

            # Freeze graph
            input_graph_path = os.path.join(FLAGS.export_dir, input_graph_name)
            input_saver_def_path = ''
            input_binary = True
            output_node_names = 'output_node'
            restore_op_name = 'save/restore_all'
            filename_tensor_name = 'save/Const:0'
            output_graph_path = os.path.join(FLAGS.export_dir, 'output_graph.pb')
            clear_devices = False
            freeze_graph.freeze_graph(input_graph_path, input_saver_def_path,
                                      input_binary, checkpoint_path, output_node_names,
                                      restore_op_name, filename_tensor_name,
                                      output_graph_path, clear_devices, '')
b)          output_graph_def = graph_util.convert_variables_to_constants(session, session.graph.as_graph_def(), ['output_node'])
        with gfile.FastGFile('./data/ldc93s1/output_graph2.pb', 'wb') as f:
            f.write(output_graph_def.SerializeToString())

but for both the dump I get the following error: -
'rnn/while/multi_rnn_cell/cell_0/gru_cell/gates/r/cond/rnn/while/multi_rnn_cell/cell_0/gru_cell/gates/r/strided_slice/_assign/RefEnter': Input tensor 'rnn/multi_rnn_cell/cell_0/gru_cell/gates/r/pop_mean:0' **Cannot convert a tensor of type float32 to an input of type float32_ref**

Any solution so far? 
I followed the similar bug but that is specifically related to Batch norm #3628 . And what is the reason behind this??

**Comment by rohan100jain** [comment]
Seems like this is a duplicate of the issue you mentioned: https://github.com/tensorflow/tensorflow/issues/3628

Seems like there is some bug in freeze_graph that is causing this issue. Maybe you should add on to that bug with a proper stack trace to debug.

**Comment by prashantmaheshwari94** [comment]
I tried to storing the model from other method which doesn't use freeze_graph. Method mentioned in 'b point' in my original post. In that also I get the same error.

**Comment by luckynarang123** [comment]
same error here

**Comment by Rajamohanreddyai** [comment]
Hello all, just follow the below video and export your own model with in a 10 seconds

https://youtu.be/w0Ebsbz7HYA
